### PR TITLE
Integrate/http methods

### LINKED ISSUE #184: Integrate gRPC Gateway for HTTP/RPC
https://github.com/grpc-ecosystem/grpc-gateway

Check examples/internal directory

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202528620406598/1202534227323526) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by prdlk**
* Utilized the gin framework to serve an HTTP API which exposes cosmos gRPC methods
* Added swagger file with API method definitions **WIP**
* Serve Swagger UI on HTTP API with Gin

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

File: highway/client/cosmos.go [hunk 1]
```diff
@@ -137,6 +137,23 @@ func (cc *Cosmos) BroadcastUpdateName(msg *rt.MsgUpdateName) (*rt.MsgUpdateNameR
 	return respMsg, nil
 }
 
+// ExistsName checks if a name exists on the blockchain
+func (cc *Cosmos) ExistsName(name string) bool {
```
Comment (mcjcloud) [inline highway/client/cosmos.go]: nit: `NameExists` is clearer.

same below

File: highway/client/webauthn.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+type WebAuthn struct {
+	instance *webauthn.WebAuthn
+
+	ctx      context.Context
+	cache    *cache.Cache
+	config   *config.Config
+	sessions *session.Store
+}
+
+// NewCosmos creates a Sonr Blockchain client with the given account and provides helper functions
```
Comment (mcjcloud) [inline highway/client/webauthn.go]: nit: update comment

File: highway/client/webauthn.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+	// store session data as marshaled JSON
+	err = wan.sessions.SaveWebauthnSession("authentication", sessionData, r, w)
+	if err != nil {
+		return nil, err
+	}
+	return options, nil
+}
+
+// SaveLoginSession saves the login session for the given user
```
Comment (mcjcloud) [inline highway/client/webauthn.go]: nit: update comment

File: highway/crud_registry.go [hunk 4]
```diff
@@ -3,203 +3,98 @@ package highway
...
-// StartRegisterName starts the registration process for webauthn on http
-func (s *HighwayServer) StartRegisterName(w http.ResponseWriter, r *http.Request) {
-	// get username/friendly name
-	vars := mux.Vars(r)
-	username, ok := vars["username"]
-	if !ok {
-		JsonResponse(w, fmt.Errorf("must supply a valid username i.e. foo.snr"), http.StatusBadRequest)
-		return
+// RegisterNameStart starts the registration process for webauthn on http
+func (s *HighwayServer) StartRegisterName(c *gin.Context) {
```
Comment (mcjcloud) [inline highway/crud_registry.go:14]: Is there no rpc variant of these requests?

File: host/exchange/protocol.go [hunk 5]
```diff
@@ -71,7 +67,7 @@ func (p *ExchangeProtocol) Request(shareReq *motor.ShareRequest) error {
 	// }
 
 	payload := &types.Payload{
-		Owner: profile,
+		// Owner: profile,
```
Comment (mcjcloud) [inline host/exchange/protocol.go]: nit: commented out code

**Review by mcjcloud — DISMISSED** [review]

File: highway/crud_registry.go (hunk 4 above)
Comment (prdlk) [reply highway/crud_registry.go:14]: Only for RegisterName - We can make this cohesive after we create monorepo.

File: host/exchange/protocol.go (hunk 5 above)
Comment (prdlk) [reply host/exchange/protocol.go]: Updated all of the nit in lastest push @mcjcloud

**Review by mcjcloud — APPROVED** [review]

**Pull request merged**